flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1050,10 +1052,15 @@ private TableResultInternal executeInternal(
     }
 
     private TableResultInternal executeQueryOperation(QueryOperation operation) {
+        String querySql = null;
+        if (operation instanceof QuerySqlOperation) {
+            querySql = ((QuerySqlOperation) operation).getQuerySql();
+        }
         CollectModifyOperation sinkOperation = new CollectModifyOperation(operation);
         List<Transformation<?>> transformations =
                 translate(Collections.singletonList(sinkOperation));
-        final String defaultJobName = "collect";
+        final String defaultJobName =
+                StringUtils.isNullOrWhitespaceOnly(querySql) ? "collect" : querySql;
```

Comment:
Add test units to check that the job name is modified

Reply:
We could have an enumeration that would hold the type of the operation ; this field would be defined at the top of the hierarchy. Then to check the type of the operation, it is a comparison on this type rather than having to use reflection.